### ISSUE TITLE
[Feature]: Hidden columns of Tab widget

### ISSUE THREAD

**Issue by clairearondel** [appsmithorg/appsmith#33857]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

 I'd like to make a suggestion for the Table widget. I have a MultiSelect widget that allows the user to select the columns they want to look at. I also have conditional formatting for some columns. If the columns were unchecked in the MultiSelect, I'd lose the conditional formatting and lose the column order (which I didn't want). So I found a way to hide the unchecked columns in the MultiSelect with the following line: {{ MultiSelect1.value.includes('name_column') }} in the parameters of each column in the 'Visible' JS. 
![appsmith_suggestion](https://github.com/appsmithorg/appsmith/assets/133198989/4c93c915-b7fd-4c19-a078-bc063083a3c7)
That way, you save space and conditional formatting! However, the only problem is that I have to do this for all 67 columns. 

### Why should this be worked on?

Perhaps it would be possible to create a main parameter for all the columns in the widget table? This would avoid the need for this line of code per column, especially if there are a lot of them. A search bar would also be interesting in cases where there are a lot of columns.

**Comment by Nikhil-Nandagopal** [comment]
@clairearondel that's a pretty unique use case! Can you tell us about the app you are building? What does it do for your organisation? Which users in your organisation want to use it?

**Comment by clairearondel** [comment]
I work in a naturalist association as a geomatician, in particular on the creation of an interface (with Appsmith) to enable naturalist experts to validate data manually on a regional scale as part of the SINP in France (data that we later send to GBIF on a global level). To structure naturalist data, we have a lot of fields to fill in. I've provided default columns that the experts will look at all the time. Sometimes, certain data will be strange and they'll want to look in more detail. Selecting only the columns that interest them will allow them to find the information they want immediately, rather than scrolling through all the columns.

In fact, we may be using Appsmith for other cases in our organisation :)

**Comment by Nikhil-Nandagopal** [comment]
@clairearondel why don't you consider connecting the table selected row to a JSON form or creating a drilldown view so that they can see all the values a lot easier? That way you can also easily provide the option to filter only the columns to see but they would be shown at the side of the table instead of inside the table. Here's an example of what that could look like. You can then add filters on top of this to hide / show certain fields
<img width="965" alt="Screenshot 2024-05-31 at 4 07 12 PM" src="https://github.com/appsmithorg/appsmith/assets/3897254/0f53e8c4-1856-4564-ab98-71c4f64b5cb2">

**Comment by clairearondel** [comment]
Yes, I'd thought about it too before doing what I did, but we wanted to keep certain functionalities of our tools so as not to lose our users. Currently, we have a data entry tool where we can choose the columns we want to see. Thanks for the tip!

**Comment by eedwardsSGS** [comment]
I'll add a +1 on this.  Show/hide columns is a common feature on reporting apps we use and it would be great to easily bring this feature to Appsmith implementations without having to edit each column through the UI.  The JSON form only works if the user only wants to see one at a time, not comparing across a number of rows.  Even using a setter function would work: tableName.columnName.setVisibility(false) like we have access to for the entire widget.

Image from another app we use regularly that we're slowly moving functionality to Appsmith for:

![image](https://github.com/appsmithorg/appsmith/assets/123981587/620865d4-ac5b-4351-bee8-cd4a8f0aac09)
